homeassistant/components/freebox/binary_sensor.py

```diff
@@ -33,21 +35,105 @@
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
-    """Set up the binary sensors."""
+    """Set up binary sensors."""
     router: FreeboxRouter = hass.data[DOMAIN][entry.unique_id]
 
     _LOGGER.debug("%s - %s - %s raid(s)", router.name, router.mac, len(router.raids))
 
-    binary_entities = [
+    binary_entities: list[BinarySensorEntity] = [
         FreeboxRaidDegradedSensor(router, raid, description)
         for raid in router.raids.values()
         for description in RAID_SENSORS
     ]
 
+    for node in router.home_devices.values():
+        if node["category"] == FreeboxHomeCategory.PIR:
+            binary_entities.append(FreeboxPirSensor(hass, router, node))
+        elif node["category"] == FreeboxHomeCategory.DWS:
+            binary_entities.append(FreeboxDwsSensor(hass, router, node))
+
+        for endpoint in node["show_endpoints"]:
+            if (
+                endpoint["name"] == "cover"
+                and endpoint["ep_type"] == "signal"
+                and endpoint.get("value") is not None
+            ):
+                binary_entities.append(FreeboxCoverSensor(hass, router, node))
+
     if binary_entities:
         async_add_entities(binary_entities, True)
 
 
+class FreeboxHomeBinarySensor(FreeboxHomeEntity, BinarySensorEntity):
+    """Representation of a Freebox binary sensor."""
+
+    _sensor_name = "trigger"
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
```

Comment:
Side note: 
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.